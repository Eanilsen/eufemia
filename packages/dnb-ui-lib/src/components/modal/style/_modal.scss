/*
 * Modal component
 *
 */

[data-dnb-modal-active='true'] {
  user-select: none;
}

#dnb-modal-root {
  position: absolute;
  top: 0;
  left: 0;
}

.dnb-modal {
  display: inline-flex;

  &__content {
    position: fixed;
    top: 0;
    left: 0;
    z-index: 3001;

    display: flex;
    align-items: center;
    justify-content: center;

    width: 100%;
    height: 100%;

    user-select: none;

    &__inner {
      position: relative;

      padding: 2rem;
      max-width: calc(100vw - 1rem);
      max-height: 100vh;
      width: auto;
      height: auto;

      overflow-y: scroll;

      user-select: text;
      border: none;

      background: var(--color-white);

      @include allAbove(medium) {
        padding: 2rem 3rem;
        padding-right: 4rem;
        @include defaultDropShadow();
      }

      @include allBelow(medium) {
        max-width: 100%;
        width: 100%;
        height: 100%;
      }

      @include allBelow(small) {
        padding: 2rem 1rem;
      }
    }
  }

  &__title.dnb-h2 {
    margin: 0 0 2rem;
  }

  // no styles set yet
  // &__trigger {
  // }

  &__close-button {
    // NB: as the specificity is several levels deep: :not([dnb-is-touch]) .dnb-button--secondary:hover:not([disabled])
    // we use !important in case we define position="relative" on hover
    position: absolute;
    top: 1rem;
    right: 1rem;

    @include allAbove(small) {
      top: 2rem;
      right: 2rem;
    }
  }

  &__overlay {
    position: fixed;
    z-index: 3000;
    left: 0;
    top: 0;

    width: 100%;
    height: 100%;

    background-color: rgba(255, 255, 255, 0.85);
  }
}
