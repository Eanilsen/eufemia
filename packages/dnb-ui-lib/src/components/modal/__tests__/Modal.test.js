/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent,
  loadScss
} from '../../../core/jest/jestSetup'

import Component from '../Modal'

// just to make sure we re-run the test in watch mode due to changes in theese files
import _modal from '../style/_modal.scss' // eslint-disable-line
import dnb_modal from '../style/dnb-modal.scss' // eslint-disable-line

const props = fakeProps(require.resolve('../Modal'), {
  all: true,
  optional: true
})
props.title = 'modal_title'
props.id = 'modal_id'
props.content_id = 'modal_content_id'
props.close_title = 'close_title'

describe('Modal component', () => {
  const Comp = mount(
    <Component
      {...props}
      modal_content="unique_modal_content"
      preventSetTriggerRef={true} // we set preventSetTriggerRef to true, cause jest gives us an error
    />
  )
  Comp.setState({
    modalActive: true
  })
  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })
  it('has to have the correct title', () => {
    expect(Comp.find('h1').text()).toBe(props.title)
  })
  it('has to have the correct aria-describedby', () => {
    expect(
      Comp.find('[aria-describedby]').props()['aria-describedby']
    ).toBe(props.content_id)
  })
  it('has to have the correct role on aria-modal', () => {
    expect(Comp.find('[aria-modal]').props().role).toBe('dialog')
  })
  it('has to have a close button', () => {
    expect(
      Comp.find(`button[aria-label="${props.close_title}"]`).props()[
        'aria-label'
      ]
    ).toBe(props.close_title)
  })
  it('should validate with ARIA rules as a dialog', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('Modal scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-modal.scss'))
    expect(scss).toMatchSnapshot()
  })
})
