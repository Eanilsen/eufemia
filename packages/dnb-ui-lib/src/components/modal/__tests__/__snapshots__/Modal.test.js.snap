// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component have to match snapshot 1`] = `
<Modal
  class="class"
  className="className"
  close_title="close_title"
  content_id="content_id"
  modal_content="modal_content"
  modal_trigger_icon="modal_trigger_icon"
  modal_trigger_text="modal_trigger_text"
  modal_trigger_title="modal_trigger_title"
  on_click={[Function]}
  preventSetTriggerRef={true}
  title="title"
  type="button"
>
  <div
    className="dnb-modal className class"
    onClick={[Function]}
  >
    <span
      className="dnb-modal__trigger dnb-modal__trigger--button"
    >
      <Button
        attributes={null}
        class={null}
        className={null}
        custom_element={null}
        custom_method={null}
        disabled={false}
        href={null}
        icon="modal_trigger_icon"
        icon_position="right"
        icon_size={null}
        id={null}
        innerRef={null}
        on_click={[Function]}
        size={null}
        text="modal_trigger_text"
        title="modal_trigger_title"
        type="button"
        variant="secondary"
      >
        <button
          className="dnb-button dnb-button--secondary dnb-button--icon-position-right dnb-button--has-text dnb-button--has-icon"
          id={null}
          onClick={[Function]}
          onMouseOut={[Function]}
          title="modal_trigger_title"
          type="button"
        >
          <Content
            attributes={null}
            class={null}
            className={null}
            content={null}
            custom_element={null}
            custom_method={null}
            disabled={false}
            href={null}
            icon="modal_trigger_icon"
            icon_position="right"
            icon_size={null}
            id={null}
            innerRef={null}
            on_click={[Function]}
            size={null}
            text="modal_trigger_text"
            title="modal_trigger_title"
            type="button"
            variant="secondary"
          >
            <span
              className="dnb-button__text"
              key="button-text"
            >
              modal_trigger_text
            </span>
            <IconWithAllIcons
              alt="modal_trigger_title"
              area_hidden={true}
              attributes={null}
              class={null}
              className="dnb-button__icon"
              color={null}
              height={null}
              icon="modal_trigger_icon"
              key="button-icon"
              modifier={null}
              size={null}
              width={null}
            />
          </Content>
        </button>
      </Button>
    </span>
    <ModalRoot
      class="class"
      className="className"
      close_title="close_title"
      content_id="content_id"
      hide={[Function]}
      modal_content="modal_content"
      modal_trigger_icon="modal_trigger_icon"
      modal_trigger_text="modal_trigger_text"
      modal_trigger_title="modal_trigger_title"
      on_click={[Function]}
      preventSetTriggerRef={true}
      title="title"
      toggleOpenClose={[Function]}
      type="button"
    >
      <ModalContent
        class="class"
        className="className"
        close_title="close_title"
        content_id="content_id"
        hide={[Function]}
        modal_content="modal_content"
        modal_trigger_icon="modal_trigger_icon"
        modal_trigger_text="modal_trigger_text"
        modal_trigger_title="modal_trigger_title"
        on_click={[Function]}
        preventSetTriggerRef={true}
        title="title"
        toggleOpenClose={[Function]}
        type="button"
      >
        <div
          aria-hidden="true"
          className="dnb-modal__content"
          onClick={[Function]}
        >
          <div
            aria-hidden="true"
            aria-labelledby="title"
            aria-modal="true"
            className="dnb-modal__content__inner dnb-no-focus dnb-style"
            id="content_id"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="dialog"
            tabIndex="-1"
          >
            modal_content
            <CloseButton
              className={null}
              close_title="close_title"
              on_click={[Function]}
            >
              <Button
                attributes={null}
                class={null}
                className="dnb-modal__close-button"
                custom_element={null}
                custom_method={null}
                disabled={false}
                href={null}
                icon="close"
                icon_position="right"
                icon_size={null}
                id={null}
                innerRef={null}
                on_click={[Function]}
                size={null}
                text={null}
                title="close_title"
                type="button"
                variant="secondary"
              >
                <button
                  className="dnb-button dnb-button--secondary dnb-button--size-medium dnb-button--icon-position-right dnb-button--has-icon dnb-modal__close-button"
                  id={null}
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  title="close_title"
                  type="button"
                >
                  <Content
                    attributes={null}
                    class={null}
                    className="dnb-modal__close-button"
                    content={null}
                    custom_element={null}
                    custom_method={null}
                    disabled={false}
                    href={null}
                    icon="close"
                    icon_position="right"
                    icon_size={null}
                    id={null}
                    innerRef={null}
                    on_click={[Function]}
                    size={null}
                    text={null}
                    title="close_title"
                    type="button"
                    variant="secondary"
                  >
                    <IconWithAllIcons
                      alt="close_title"
                      area_hidden={true}
                      attributes={null}
                      class={null}
                      className="dnb-button__icon"
                      color={null}
                      height={null}
                      icon="close"
                      key="button-icon"
                      modifier={null}
                      size={null}
                      width={null}
                    >
                      <span
                        aria-hidden={true}
                        aria-label="close title"
                        className="dnb-icon has-custom-size dnb-button__icon"
                        role="img"
                      >
                        <close>
                          <svg
                            viewBox="0 0 16 16"
                          >
                            <path
                              clipRule="evenodd"
                              d="M4.03 3.1a.75.75 0 0 0-1.06 1.06l3.97 3.97-3.97 3.97a.75.75 0 1 0 1.06 1.06L8 9.19l3.97 3.97a.75.75 0 1 0 1.06-1.06L9.06 8.13l3.97-3.97a.75.75 0 0 0-1.06-1.06L8 7.07 4.03 3.1z"
                              fillRule="evenodd"
                            />
                          </svg>
                        </close>
                      </span>
                    </IconWithAllIcons>
                  </Content>
                </button>
              </Button>
            </CloseButton>
          </div>
        </div>
        <span
          aria-hidden="true"
          className="dnb-modal__overlay"
        />
      </ModalContent>
    </ModalRoot>
  </div>
</Modal>
`;

exports[`Modal scss have to match snapshot 1`] = `
"/*
* DNB Modal
*
* ======================================================================== */
/*
 * Utilities
 * ========================================================================== */
/*
 * Typography
 * ========================================================================== */
:root {
  --font-family-std: 'Fedra Sans Std', sans-serif;
  --font-family-demi: 'Fedra Sans Std Demi', sans-serif; }

.typo-std,
.typo-book {
  font-family: var(--font-family-std);
  font-weight: normal;
  font-style: normal; }

.typo-medium {
  font-family: var(--font-family-std);
  font-weight: 600;
  font-style: normal; }

.typo-demi {
  font-family: var(--font-family-demi);
  font-weight: normal;
  font-style: normal; }

.typo-number--old-style {
  font-variant-numeric: oldstyle-nums;
  -moz-font-feature-settings: \\"onum\\";
  -webkit-font-feature-settings: \\"onum\\";
  font-feature-settings: \\"onum\\"; }

.typo-number--lining {
  font-variant-numeric: lining-nums;
  -moz-font-feature-settings: \\"lnum\\";
  -webkit-font-feature-settings: \\"lnum\\";
  font-feature-settings: \\"lnum\\"; }

.typo-number--tabular {
  font-variant-numeric: tabular-nums;
  -moz-font-feature-settings: \\"tnum\\";
  -webkit-font-feature-settings: \\"tnum\\";
  font-feature-settings: \\"tnum\\"; }

.typo-number--proportional {
  font-variant-numeric: proportional-nums;
  -moz-font-feature-settings: \\"pnum\\";
  -webkit-font-feature-settings: \\"pnum\\";
  font-feature-settings: \\"pnum\\"; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std Demi';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

/*
* Scopes
* ======================================================================== */
/*
* DNB Button
*
* ======================================================================== */
/*
* DNB icon
*
* ======================================================================== */
/*
* Icon component
*
* ======================================================================== */
.dnb-icon {
  display: inline;
  vertical-align: baseline;
  font-size: inherit;
  width: 1em;
  height: 1em; }
  .dnb-icon svg {
    fill: currentColor; }
  .dnb-icon:not(.has-custom-size) svg {
    width: inherit;
    height: inherit; }
  .dnb-icon--small {
    width: 0.5em;
    height: 0.5em; }
  .dnb-icon--default {
    width: 1em;
    height: 1em; }
  .dnb-icon--medium {
    width: 1.5em;
    height: 1.5em; }
  .dnb-icon--large {
    width: 2em;
    height: 2em; }

.dnb-button {
  font-family: var(--font-family-std, sans-serif);
  font-size: 1rem;
  font-weight: normal;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333); }
  .dnb-button *,
  .dnb-button ::before,
  .dnb-button ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-button ::before,
  .dnb-button ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
* Button component
*
* ======================================================================== */
:root {
  --button-font-family: var(--font-family-std);
  --button-font-size: 1rem;
  --button-font-weight: normal;
  --button-font-style: normal;
  --button-width: 2.5rem;
  --button-height: 2.5rem;
  --button-width--medium: 2rem;
  --button-height--medium: 2rem;
  --button-width--large: 3rem;
  --button-height--large: 3rem;
  --button-icon-size: 1rem;
  --button-border-width: 1px;
  --button-border-width--hover: 3px;
  --button-border-radius: calc(var(--button-height) / 2);
  --button-border-radius--medium: calc(var(--button-height--medium) / 2);
  --button-border-radius--large: calc(var(--button-height--large) / 2); }

.dnb-button {
  display: inline-flex;
  vertical-align: middle;
  align-content: center;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
  cursor: pointer;
  width: var(--button-width);
  height: var(--button-height);
  padding: 0;
  border: var(--button-border-width) solid transparent;
  border-radius: var(--button-border-radius);
  font-family: var(--button-font-family);
  font-size: var(--button-font-size);
  font-weight: var(--button-font-weight);
  font-style: var(--button-font-style);
  text-decoration: none;
  /* Firefox includes a hidden border which messes up button dimensions */ }
  .dnb-button__icon {
    line-height: 0; }
    .dnb-button__icon:not(.dnb-icon) svg:not([width]):not([height]) {
      width: var(--button-icon-size);
      height: var(--button-icon-size); }
  .dnb-button__text {
    color: inherit;
    background-color: inherit;
    transform: translateY(var(--button-border-width)); }
  .dnb-button--size-medium {
    width: var(--button-width--medium);
    height: var(--button-height--medium);
    border-radius: var(--button-border-radius--medium); }
  .dnb-button--has-text.dnb-button--size-medium {
    padding-left: 1em;
    padding-right: 1em; }
  .dnb-button--has-text.dnb-button--icon-position-left.dnb-button--size-medium {
    padding-left: 0.5em; }
  .dnb-button--has-text.dnb-button--icon-position-right.dnb-button--size-medium {
    padding-right: 0.5em; }
  .dnb-button--size-large {
    width: var(--button-width--large);
    height: var(--button-height--large);
    border-radius: var(--button-border-radius--large); }
  .dnb-button--has-text.dnb-button--size-large {
    padding-left: 2em;
    padding-right: 2em; }
  .dnb-button--has-text.dnb-button--icon-position-left.dnb-button--size-large {
    padding-left: 1em; }
  .dnb-button--has-text.dnb-button--icon-position-right.dnb-button--size-large {
    padding-right: 1em; }
  .dnb-button--has-text {
    padding-left: 1.5em;
    padding-right: 1.5em;
    width: auto; }
  .dnb-button--has-text .dnb-button__icon {
    margin: 0 calc(var(--button-icon-size) / 2); }
  .dnb-button--has-text.dnb-button--icon-position-left {
    padding-left: 0.5em; }
  .dnb-button--has-text.dnb-button--icon-position-right {
    padding-right: 0.5em; }
  .dnb-button--has-text.dnb-button--has-icon .dnb-button__icon {
    order: 2; }
  .dnb-button--has-text.dnb-button--has-icon .dnb-button__text {
    order: 1; }
  .dnb-button--has-text.dnb-button--has-icon.dnb-button--icon-position-left .dnb-button__icon {
    order: 1; }
  .dnb-button--has-text.dnb-button--has-icon.dnb-button--icon-position-left .dnb-button__text {
    order: 2; }
  .dnb-button--reset {
    margin: 0;
    padding: 0;
    width: auto;
    height: auto;
    overflow: visible;
    border: none;
    border-radius: 0;
    background-color: transparent;
    appearance: none;
    box-shadow: none;
    color: inherit;
    font: inherit;
    text-align: inherit;
    line-height: 0; }
    .no-touch .dnb-button--reset:hover[disabled] {
      cursor: not-allowed; }
    .no-touch .dnb-button--reset:hover:not([disabled]) {
      opacity: 0.6;
      box-shadow: none;
      border: none; }
    .dnb-button--reset:not([disabled]):focus, .dnb-button--reset:not([disabled]):active {
      outline: none;
      border-color: transparent; }
      html[data-whatinput='mouse'] .dnb-button--reset:not([disabled]):focus:not(.dnb-no-mouse-focus), html[data-whatinput='mouse'] .dnb-button--reset:not([disabled]):active:not(.dnb-no-mouse-focus) {
        transition: box-shadow 0.1s ease-out 0s;
        box-shadow: 0 0 0 2px var(--color-emerald-green); }
      html[data-whatinput='keyboard'] .dnb-button--reset:not([disabled]):focus, html[data-whatinput='keyboard'] .dnb-button--reset:not([disabled]):active {
        box-shadow: 0 0 0 2px var(--color-signal-orange); }
    html[data-whatinput='mouse']
.dnb-button--reset:not([disabled]):focus:not(.dnb-no-mouse-focus),
    html[data-whatinput='mouse']
.dnb-button--reset:not([disabled]):active:not(.dnb-no-mouse-focus) {
      box-shadow: none;
      color: inherit;
      border: none; }
  .dnb-button[type='button'] {
    appearance: none; }
  a .dnb-button {
    border-bottom: none; }
  button.dnb-button::-moz-focus-inner .dnb-button {
    border: none; }

.dnb-modal {
  font-family: var(--font-family-std, sans-serif);
  font-size: 1rem;
  font-weight: normal;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333); }
  .dnb-modal *,
  .dnb-modal ::before,
  .dnb-modal ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-modal ::before,
  .dnb-modal ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
 * Modal component
 *
 * ========================================================================== */
body[data-modal-active='true'] {
  overflow: hidden; }

#dnb-modal-root {
  position: absolute;
  top: 0;
  left: 0; }

.dnb-modal {
  display: inline-flex; }
  .dnb-modal__content {
    position: fixed;
    top: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    user-select: none;
    z-index: 3001; }
    .dnb-modal__content__inner {
      position: relative;
      user-select: text;
      padding: 2rem 1rem;
      max-width: calc(100vw - 1rem);
      border: none;
      box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.15);
      background: white; }
      @media screen and (min-width: 700px) {
        .dnb-modal__content__inner {
          max-width: 500px;
          padding: 2rem 3rem; } }
  .dnb-modal__trigger {
    cursor: pointer;
    user-select: none; }
  .dnb-modal__trigger--text:not([class*='tertiary']) {
    border: none;
    padding: none;
    background: transparent; }
    .dnb-modal__trigger--text:not([class*='tertiary']):hover {
      text-decoration: underline; }
  .dnb-modal__close-button, .dnb-modal__close-button.dnb-button--tertiary {
    position: absolute;
    top: 1rem;
    right: 1rem;
    cursor: pointer;
    color: var(--color-sea-green); }
  .dnb-modal__overlay {
    position: fixed;
    z-index: 3000;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background: rgba(255, 255, 255, 0.85); }
"
`;
